modules/core/src/minmax.cpp

```diff
@@ -1510,9 +1510,19 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
 
     Mat src = _src.getMat(), mask = _mask.getMat();
 
+    int _minIdx, _maxIdx;
+    int* min_offset = (cn == 1) ? minIdx : &_minIdx;
+    int* max_offset = (cn == 1) ? maxIdx : &_maxIdx;
```

Comment:
1. This seems like a bug because as documented "When minIdx is not NULL, it must have at least 2 elements"
2. Passing a non-null pointer means the HAL implementation can't optimise for minIdx and maxIdx being null.
I suggest instead passing minIdx and maxIdx directly to cv_hal_minMaxIdx at line 1519. We already know from the assert above that they must be null if cn>1.

Reply:
Agreed.